provisioning/README.md

```diff
@@ -0,0 +1,109 @@
+# Mbed Crypto provisioning example application
+
+Let's say the secure element manufacturer shipped you a secure element with a
+device private key already inside. This type of "pre-provisioned" secure
+element is what you'll use as a part during the device assembly process. You've
+taken this secure element, paired it with a nice Cortex-M microcontroller on a
+board to make a device. Now, you'd like to be able to use this key from Mbed
+Crypto. How does Mbed Crypto know the key exists? We can tell Mbed Crypto by
+running a provisionining application, which is typically separate from the
+primary application and may be run as a final device assembly step.
```

Comment:
This paragraph does not do a good job of answering the fundamental introductory questions: what is this about? why should I care? Is this even applicable to me?

> Let's say the secure element manufacturer

Which secure element manufacturer? The audience is application developers. Their focus is on the device as a whole, not on one particular hardware feature.

> You've taken this secure element, paired it with a nice Cortex-M microcontroller on a board to make a device

In many cases, someone has already done that for me: I bought the hardware or the hardware design, and it has a secure element next to the main chip or inside the main chip.

> How does Mbed Crypto know the key exists? We can tell Mbed Crypto by running a provisionining application

A key provisioning application would import a key, not register an existing key. The code here registers an existing key and generates and uploads a new one.

> a provisionining application, which is typically separate from the primary application and may be run as a final device assembly step.

The code that creates and registers the bootstrap key is a device assembly step, but that's not what this example does. The code that uses the bootstrap key to register a production key, which is what this example does, is not an assembly step.

Reply:
Here's how I'd write the introduction.

> The [Microchip ATECC608A](https://www.microchip.com/wwwproducts/en/ATECC608A) is a secure element which can store and use private keys that cannot be extracted. Secure element manufacturers usually create an initial private key as part of the manufacturing process. You can create a new private key as part of the manufacturing process of the whole device, or as part of its deployment. This example application demonstrates how to register an existing key and generate a new key through Mbed Crypto. This application also demonstrates how to create a certificate signing request (CSR) for the keys through Mbed TLS. Once you have a CSR, you can send it to your certificate authority (CA) to obtain a certificate which you'll then use to authenticate the device.

Reply:
"The audience is application developers." No. Ordinary Mbed OS application developers should not be doing the device manufacturing. The audience is device manufacturers, people assembling boards and making sure they ship with their firmware in a good state: ready for the ordinary Mbed OS application developer to just run the `mutual-tls` example app or similar.